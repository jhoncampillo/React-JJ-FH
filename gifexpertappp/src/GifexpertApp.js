import React from "react";
import { useState } from "react";
import AddCategory from "./components/AddCategory";
import GifGrid from "./components/GifGrid";

const GifexpertApp = ({ defaultCategories = [] }) => {
  // const categories = ["OnePunch", "SamurayX", "Dragon Ball"];
  //le envio el  parametro por default al state
  const [categories, setCategories] = useState(defaultCategories);
  //const [categories, setCategories] = useState(["One Punch"]);

  //   const handleAdd = () => {
  //     //De ambas formas  se actualiza el Array de catgories
  //     //setCategories([...categories, "HunterXHunter"]);
  //     setCategories((cats) => [...cats, "HunterXHunters"]);
  //   };

  return (
    <div>
      <h1>GifExpertApp</h1>
      {/* le mando el parametro setcategories al compoenete por referencia 
      para poder cambier ael estado y adicionra categories */}
      <AddCategory setCategories={setCategories} />
      <hr />
      {/* <button onClick={handleAdd}>Agregar</button> */}

      <ol>
        {/* //recorro el estado inicial de array Usestate */}
        {categories.map((category) => (
          //<li key={category}> {category}</li> // lo remplazo por el Componente GifGrid
          <GifGrid key={category} category={category} />
        ))}
        {/* //mando categories del useState l componente GifGrid
         // referencio los id con key que es necesario - No puede ser el
        index y debe de ser unico
        //como es una funcion de flecha que solo regresa un elemento 
        // se puede obiar la llave de arriba  quito el return y envolver en parentesis
        //para indicar esta retornndo un objeto
        // return <li key={category}> {category}</li> */}
      </ol>
    </div>
  );
};

export default GifexpertApp;
