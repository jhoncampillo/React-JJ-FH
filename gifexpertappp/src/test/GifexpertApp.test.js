import React from "react";
import { shallow } from "enzyme";
import GifexpertApp from "../GifexpertApp";

describe("Pruebas al componente GitexpertApp", () => {
  test("Debe hacer match snapshot  ", () => {
    //creo el snapshot
    const wrapper = shallow(<GifexpertApp />);

    //espero que el snapchot haga match con el wrapper
    expect(wrapper).toMatchSnapshot();
  });

  //segunda prueba que dbe de mostarar una lista de categorias

  test("Debe de mostrar lista de categorias", () => {
    const categories = ["One Punch", "Dragon Ball"];
    const wrapper = shallow(<GifexpertApp defaultCategories={categories} />);

    //evalue que haha match con el snapshot
    expect(wrapper).toMatchSnapshot();
    //evalue que el contenido del  GitGrid sea del mismo tamano que  el array de  las categories
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
