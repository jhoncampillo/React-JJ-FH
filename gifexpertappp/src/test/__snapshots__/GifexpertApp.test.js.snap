// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente GitexpertApp Debe de mostrar lista de categorias 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</div>
`;

exports[`Pruebas al componente GitexpertApp Debe hacer match snapshot   1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;
